### ISSUE TITLE
New version overwrite any existing favorites

### ISSUE THREAD

**Issue by johnpapa** [johnpapa/vscode-peacock#236]
It should merge them. Tests are in adequate or missing a scenario.

**Comment by johnpapa** [comment]
I've seen this happen, though I am yet to reproduce it to determine the cause.

I added tests to the "adding recommended favorites" command to make sure it doesn't remove any new ones a user may have set.

Next step is to see what other code may be causing this.

if anyone else can repro this, details are appreciated.

**Comment by attomos** [comment]
I've noticed this issue with my main VS Code, just downgraded to 2.0.0 to keep the existing Peacock configs.

But I can reproduce this with newly installed VS Code Insiders as well.

### Steps to reproduce
1. Start fresh instance of VS Code Insiders with no settings/extensions
2. Install Peacock 2.0.0
3. Run `peacock.changeColorToRandom`. It works fine with these **configs**:
```jsonc
// settings.json
{
    "telemetry.enableTelemetry": false,
    "workbench.colorTheme": "Visual Studio Light",
    "peacock.affectStatusBar": true,
}
```
```jsonc
//  .vscode/settings.json
{
    "workbench.colorCustomizations": {
        "titleBar.activeBackground": "#f48118",
        "titleBar.inactiveBackground": "#f4811899",
        "titleBar.activeForeground": "#15202b",
        "titleBar.inactiveForeground": "#15202b99",
        "statusBar.background": "#f48118",
        "statusBarItem.hoverBackground": "#cf680a",
        "statusBar.foreground": "#15202b"
    }
}
```
4. Update Peacock to 3.0.1 and my workspace settings changed to
```jsonc
//  .vscode/settings.json
{
}
```

**Comment by johnpapa** [comment]
Thanks. This is slightly different where workspace colors are removed. I have code that saves them from 2.5 to 3.0. But from 2.0 to 3.0 (large jump) I have no clear way to keep them. 

Thanks though for sharing 


The issue listed here is where you add a favorite color of your own and o have seen it disappear at some point. Though o can’t tell if it’s just me in testing or it is removed

**Comment by Sata51** [comment]
I use peacock since it comes out. And I've set it up with the basic settings of the color `Angular Red`

```json
"workbench.colorCustomizations": {
        "titleBar.activeBackground": "#b52e31",
        "titleBar.inactiveBackground": "#b52e3199",
        "titleBar.activeForeground": "#e7e7e7",
        "titleBar.inactiveForeground": "#e7e7e799"
    }
```

At each start, with the `3.0.1` version, the json object `workbench.colorCustomizations` in the 
 `settings.json` is deleted.

If I set the color with the `3.0.1` version, it adds a json object `peacock.color` and didn't erase my old configuration.

If I set the color with the `3.0.1` version, and delete the unwanted configuration, It replace the configuration everytime with the full configuration.

**Comment by johnpapa** [comment]
@sata51 - please open a new issue as this is not the same as this issue. 

Also - this behavior is documented in the docs. Peacock.color setting defines the color and tells peacock what to do. Open a new issue if you need more.

**Comment by ajonp** [comment]
@Sata51 same issue here, let me know if you want me to open issue. Downgraded to 2.5 worked for now.

**Comment by Sata51** [comment]
@ajonp I found a solution, I've updated my configuration to:

```json
    "workbench.colorCustomizations": {
        "titleBar.activeBackground": "#b52e31",
        "titleBar.inactiveBackground": "#b52e3199",
        "titleBar.activeForeground": "#e7e7e7",
        "titleBar.inactiveForeground": "#e7e7e799"
    },
    "peacock.affectActivityBar": false,
    "peacock.affectStatusBar": false,
    "peacock.color": "#b52e31"
```

This pattern match my usage

**Comment by johnpapa** [comment]
all - Please understand that there are 2 different things being discussed here. The issue is for the Peacock favorites overwriting your favorites.

There are also people talking about their colors changing. The colors changing is due to a breaking change in v3 (please read the changelog for more details). 

Let's keep this issue about the favorites set of colors only. If you have a different issue, please open one.

**Comment by johnpapa** [comment]
To help clarify, I'm removing comments that are not relevant to this specific issue.

**Comment by johnpapa** [comment]
closed by #252
